### PR TITLE
Dialog design improvement

### LINKED ISSUE #1011: Cancel button needed on add bookmark
If I right click and choose **Add Bookmark**, there is no visible cancel button. To cancel it requires to click on any empty place, which is somewhat confusing I think. So a cancel button will be comfortable.

### PR DESCRIPTION
**Pull request by luixxiul**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Ran `git rebase -i` to squash commits if needed.

This improves styling inside .genericFormTable.

### CODE REVIEW

File: less/button.less [hunk 1]
```diff
@@ -8,7 +8,7 @@ span.browserButton {
   cursor: default;
   display: inline-block;
   line-height: 30px;
-  width: 30px;
+  width: 40px;
```
Comment (luixxiul) [inline less/button.less]: Increased to centralize "Cancel", still I think this is not the best solution concerning l10n.

File: less/forms.less [hunk 2]
```diff
@@ -277,26 +277,34 @@
...
+        justify-content: flex-end;
+        margin: 2px;
       }
-      >label {
-        padding-right: 5px;
-       text-align: right;
+
+      > label {
+        padding-right: 4px;
+        width: 20%;
```
Comment (luixxiul) [inline less/forms.less]: This could be 30%, which seems to me too wide though.

File: js/components/addEditBookmark.js [hunk 3]
```diff
@@ -115,8 +115,8 @@ class AddEditBookmark extends ImmutableComponent {
             </select>
           </div>
           <div className='formRow'>
-            <span />
             <Button l10nId='save' className='primaryButton' onClick={this.onSave} />
+            <Button l10nId='cancel' className='subtleButton cancelButton' onClick={this.onClose} />
```
Comment (luixxiul) [inline js/components/addEditBookmark.js]: I'm not sure if this is the optimal order from a viewpoint of UX. "Save" "Cancel" or "Cancel" "Save"?

File: js/components/loginRequired.js [hunk 4]
```diff
@@ -94,8 +94,8 @@ class LoginRequired extends React.Component {
             : null
           }
           <div className='formRow'>
-            <span />
             <Button l10nId='ok' className='primaryButton' onClick={this.onSave.bind(this)} />
+            <Button l10nId='cancel' className='subtleButton cancelButton' onClick={this.onClose} />
```
Comment (luixxiul) [inline js/components/loginRequired.js]: ditto

File: js/components/addEditBookmark.js [hunk 5]
```diff
@@ -115,8 +115,8 @@ class AddEditBookmark extends ImmutableComponent {
             </select>
           </div>
           <div className='formRow'>
-            <span />
```
Comment (luixxiul) [inline js/components/addEditBookmark.js]: This is no longer required thanks to display:flex.

**Comment by bbondy** [comment]
@bradleyrichter for design review

**Comment by bradleyrichter** [comment]
Button order needs to be os specific. I'll look at other ideas to remove "cancel."

**Comment by luixxiul** [comment]
I see, pushing again without the button for now.

File: less/forms.less [hunk 6]
```diff
@@ -277,26 +277,30 @@
   }
 
   .genericFormTable {
-    display: table;
-    >div {
+    display: flex;
+    flex-flow: column nowrap;
+    min-width: 350px;
```
Comment (luixxiul) [inline less/forms.less]: This could be less than that.

**Comment by bradleyrichter** [comment]
@bbondy This is fine to merge. Thanks guys!

**Comment by bbondy** [comment]
++

**Pull request merged**